src/sentry/event_manager.py

```diff
@@ -717,6 +717,8 @@ def _get_event_instance(self, project_id=None):
         # unused
         time_spent = data.pop('time_spent', None)
 
+        data['node_id'] = '{}:{}'.format(project_id, event_id)[:40]
```

Comment:
I'm a little nervous to do this since it does remove uniqueness, especially as project_ids get larger, you get less and less unique.

Reply:
Since event_id is a 32-char hex string, it really only encodes 16 bytes of info, so I can compress it into 16 bytes, + base64 penalty of 1.5 gives 24 bytes. which is the same length as the current `b64encode(uuid4().bytes)` solution. That would leave enough space for project_ids to to to 1 quadrillion before truncation :)

But `project_id:b64encode(compressed_event_id)` is a bit more obscure and less grokkable than `project_id:event_id` so there is a little bit of a tradeoff I guess?

Reply:
I'm just remembering now that maybe event_id isn't strictly hex as it's case sensitive?, so like, `[a-fA-F0-9]` is a little more than 4 bits per char